src/TextureAsset.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <string>
+
+#include "GL/glew.h"
+#include "SDL_opengl.h"
+
+class TextureAsset {
+public:
+    TextureAsset();
+    ~TextureAsset();
+    bool load_texture(const std::string& file_path);
+
+    bool valid_;
```

Comment:
By making these public, other classes have freedom to change these as they will. They would be beteer as private with a getter.